templates/CMakeLists.txt.template

```diff
@@ -522,7 +522,7 @@
     PRIVATE <%text>${CARES_INCLUDE_DIR}</%text>
     PRIVATE <%text>${CMAKE_CURRENT_BINARY_DIR}</%text>/third_party/cares/cares
     PRIVATE <%text>${CMAKE_CURRENT_BINARY_DIR}</%text>/third_party/gflags/include
-    PRIVATE <%text>${CMAKE_CURRENT_BINARY_DIR}</%text>/third_party/abseil-cpp
+    PRIVATE <%text>${CMAKE_CURRENT_SOURCE_DIR}</%text>/third_party/abseil-cpp
```

Comment:
the cares and gflags under `${CMAKE_CURRENT_BINARY_DIR}` is actually for headers that are generated during cares and gflags build.   source_dir is the right place to include (same as for other third_party headers).